src/pages/index.js

```diff
@@ -1,120 +1,267 @@
-const editFormValidator = new FormValidator(config, profileEditForm);
+const addFormValidator = new FormValidator(config, addCardForm);
+const editFormValidator = new FormValidator(config, profEditForm);
+const avatEditFormValidator = new FormValidator(config, avatEditForm);
 
 /* --------------------- Card Handler --------------------- */
 
-// Create image Popup instance.
-const cardPreviewPopup = new PopupWithImage(selectors.previewPopup);
+let cardSection;
+let card;
 
-// This function creates a new card
-function createCard(cardData) {
-  const cardElement = new Card(
-    {
-      cardData,
-      handleImageClick: (imageData) => {
-        cardPreviewPopup.open(imageData);
-      },
-    },
-    selectors.cardTemplate
-  );
+/* --------------- Edit Avatar Card --------------- */
 
-  return cardElement.getView();
+// Edit profile Avatar thumbnail imageData {avatURL}
+const editAvatPopup = new PopupWithForm(
+  selectors.editAvatFormModal,
+  handleAvatarSubmit,
+  "Save",
+  "Saving..."
+);
+
+function handleAvatarSubmit(imageData) {
+  editAvatPopup.showLoading();
+  api
+    .setAvatarAPI(imageData.avatURL)
+    .then((res) => {
+      userInfo.setAvatar(imageData.avatURL);
+      editAvatPopup.close();
+    })
+    .finally(() => {
+      editAvatPopup.hideLoading();
+    });
 }
 
-// Create a section of cards
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      // Create a new card
-      const cardElement = createCard(cardData);
+editAvatButton.addEventListener("click", () => {
+  // Open the avatar edit modal form
+  editAvatPopup.open();
+  // Reset validation for the avatar edit modal form
+  avatEditFormValidator.resetForm();
+});
 
-      // Display each card
-      cardSection.addItem(cardElement);
-    },
-  },
-  selectors.cardsList
-);
-cardSection.renderItems(initialCards);
+editAvatPopup.setEventListeners();
 
 /* ------------------- Profile Info ------------------- */
 
-const userInfo = new UserInfo(
-  selectors.profileName,
-  selectors.profileProfession
+const userInfo = new UserInfo({
+  profNameSelector: selectors.profName,
+  profAboutSelector: selectors.profAbout,
+  profAvatarSelector: selectors.profAvatar,
+});
+
+const editProfInfo = new PopupWithForm(
+  selectors.editProfFormModal,
+  handleProfEditSubmit,
+  "Save",
+  "Saving..."
 );
-/* --------------------- Edit Card -------------------- */
 
-// Create the edit form instance
-const editFormPopup = new PopupWithForm(selectors.editFormPopup, (values) => {
-  // Add the form's input to the profile section
-  userInfo.setUserInfo(values.name, values.description);
-});
+function handleProfEditSubmit({ name, about }) {
+  editProfInfo.showLoading();
+  api
+    .setProfileInfoAPI(name, about)
+    .then((dataAPI) => {
+      userInfo.setProfileInfo({
+        name: name,
+        about: about,
+      });
+
+      editProfInfo.close();
+    })
+    .finally(() => {
+      editProfInfo.hideLoading();
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```